LoL_PlayerEventGenertor/src/main/java/player/Room.java

```diff
@@ -0,0 +1,129 @@
+package player;
+
+import java.time.OffsetDateTime;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.IntStream;
+
+public class Room implements Runnable{
+    static int userNum;
+    static int player = 10;
+    private final String sessionRoomID;
+    static OffsetDateTime createRoomDate;
+
+    static Set<String> ipSet = new HashSet<>();
+    static Set<String> accSet = new HashSet<>();
+    static Set<Integer> championSet = new HashSet<>();
+    static Random rand = new Random();
+
+    public Room(String sessionRoomId, OffsetDateTime createRoomDate, int userNum) {
+        this.sessionRoomID = sessionRoomId;
+        this.createRoomDate = createRoomDate;
+        this.userNum = userNum;
+    }
+
+    CountDownLatch latch = new CountDownLatch(player);
+    ExecutorService executor = Executors.newFixedThreadPool(player);
```

Comment:
앞서 Main에서 쓰레드 풀을 생성한 다음, 내부에서 다시 새로운 쓰레드 풀을 만들고 있는 것으로 보입니다.
위와 같은 구조에서는 자원이 낭비될 수 있고 적절히 관리되지 않으면 메모리 누수나 데드락이 발생할 수 있을 것 같아요.
상위에서 생성된 쓰레드 풀을 재사용하는건 어떨까요?

```suggestion
    private final ExecutorService executorService;

    public Room(..., ExecutorService executorService) {
        ...
        this.executorService = executorService;
    }
    CountDownLatch latch = new CountDownLatch(player);
```

Reply:
Main에서 Room 쓰레드 풀을 생성하고, Room에서 10개의 Player 쓰레드 풀을 생성합니다.
1개의 Room에는 10명의 플레이어가 존재합니다.
- Room
  - Player
  - Player
  -   ....

Main에 있는 Room 쓰레드 풀을 재사용하게 되면 10개의 플레이어 쓰레드 풀을 생성하지 못하는 거 같습니다!

Reply:
Room에서 쓰레드 풀 사이즈를 늘려서 사용하는 방법을 생각했는데 격리해야한다면 지금처럼 유지해도 괜찮을 것 같네요.
보통 작업의 성격이나 수명주기가 다른 경우에 격리하기도 합니다 👍🏻️